examples/echo_service.rs

```diff
@@ -23,7 +26,7 @@ async fn main() -> Result<(), Error> {
 
     let qp2p = QuicP2p::with_config(
         Some(Config {
-            ip: None,
+            ip: Some(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1))),
```

Comment:
Using localhost here might beat the purpose of this example. The primary goal of this example is to test the echo service using which the public IP of the connecting node is determined. 

The ideal fix for this would be to error out when the IP is not specified when calling the example with `create`. 
I can raise a PR with the fix for this and also document this in comments if that's okay?